packages/kbn-http-tools/src/types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ByteSizeValue } from '@kbn/config-schema';
+
+export interface IHttpConfig {
+  host: string;
+  port: number;
+  maxPayload: ByteSizeValue;
+  keepaliveTimeout: number;
+  socketTimeout: number;
```

Comment:
I had to create interfaces for the intersection of the properties used both by both `HttpServer` and the `BasePathProxyServer`, that are used for the inputs of the methods I extracted in this package. Core's `HttpConfig` now implements this new `IHttpConfig` interface.